HealthcareSystem/Backend/Service/SftpService/SftpService.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Backend.Model.Pharmacies;
+using Renci.SshNet;
+using Renci.SshNet.Sftp;
+
+namespace Backend.Service.SftpService
+{
+    public class SftpService : ISftpService
+    {
+        private readonly SftpConfig _config;
+
+        public SftpService()
+        {
+            _config = new SftpConfig();
+        }
+
+        public IEnumerable<SftpFile> ListAllFiles(string remoteDirectory = ".")
+        {
+            using var client = new SftpClient(_config.Host, _config.Port == 0 ? 22 : _config.Port, _config.Username, _config.Password);
+            try
+            {
+                client.Connect();
+                return client.ListDirectory(remoteDirectory);
+            }
+            catch (Exception exception)
+            {
+                Console.WriteLine(exception);
+                return null;
```

Comment:
Will this have proper error handling eventually? At least a message about failure reaching the client, since I don't think recovery is exactly possible here.

Reply:
Yeah I think when we completely figure out what we want to do with frontend we should do a PR just for error handling like this and have them reported to the user, as this is not the only place that's missing error handling.